vmas/simulator/environment/environment.py

```diff
@@ -770,6 +773,53 @@ def render(
         # render to display or array
         return self.viewer.render(return_rgb_array=mode == "rgb_array")
 
+    def plot_boundary(self):
+        # Include boundaries in the rendering
+        from vmas.simulator import rendering
+        from vmas.simulator.rendering import Line
+        from vmas.simulator.utils import Color
+
+        # Check boundary limits
+        if self.world.x_semidim is not None or self.world.y_semidim is not None:
+
+            infinite_value = 1e7
+
+            # set semidim variables
+            x_semi = (
+                self.world.x_semidim
+                if self.world.x_semidim is not None
+                else infinite_value
+            )
+            y_semi = (
+                self.world.y_semidim
+                if self.world.y_semidim is not None
+                else infinite_value
+            )
+
+            # define edges
+            boundary_edges = [
```

Comment:
In the case of only one boundary set we need just 2 lines not 4

Reply:
> In the case of only one boundary set we need just 2 lines not 4

Sorry, are you sure?

Eventhough are just two lines, in my opinion, we should specify 4 points. Anyway, I agree that the previous implementation drew a square/rectangle.

I think the point regards the for-loop for drawing lines, for this reason I suggest this new implementation. But I'm not understanding it doesn't draw nothing in the case of just "two parallel lines".

```python
if x_semi is not None and y_semi is not None: # drawing square/rectangle

    boundary_edges = [
        (-x_semi, y_semi),  # top_left
        (x_semi, y_semi),  # top_right
        (x_semi, -y_semi),  # bottom_right
        (-x_semi, -y_semi)  # bottom_left
    ]

    # Create lines to form the boundary by connecting each corner to the next
    for i in range(len(boundary_edges)):
        start = boundary_edges[i]  # Current corner point
        end = boundary_edges[
            (i + 1) % len(boundary_edges)]  # Next corner point, wrap around to the first point
        print(start, end)
        line = Line(start, end, width=0.7)  # Create a line between two corners
        line.set_color(*color)  # Set the line color
        self.viewer.add_onetime(line)  # Add the line to the viewer for rendering

else: # drawing two parallel lines

    # if only x_semi is provided, draw horizontal lines
    if x_semi is not None:
        boundary_edges = [
            (-x_semi, infinite_value),  # upper_left
            (x_semi, infinite_value),  # upper_right
            (-x_semi, -infinite_value),  # lower_left
            (x_semi, -infinite_value)  # lower_right
        ]

    # if only y_semi is provided, draw vertical lines
    elif y_semi is not None:
        boundary_edges = [
            (infinite_value, y_semi),  # right_top
            (infinite_value, -y_semi),  # right_bottom
            (-infinite_value, y_semi),  # left_top
            (-infinite_value, -y_semi)  # left_bottom
        ]

    else:
        boundary_edges = []

    # create the two parallel lines by connecting the points
    for i in range(0, len(boundary_edges), 2):
        start = boundary_edges[i]
        end = boundary_edges[i + 1]
        print(start, end)
        line = Line(start, end, width=0.7)
        line.set_color(*color)
        self.viewer.add_onetime(line)
```

What do you think?

Reply:
I did wrong in labelling `boundary_edges`, the correct name should be `boundary_points` or something like this.

Reply:
yes what you did is what i was thinking, still 4 points, but only draw 2 lines when only one semidim is available.

I do think we can write it a little more compactly tho:
- define the 4 points like before
- write only once the line creation loop making it work for 2 or 4 lines

Reply:
if it is too difficult also as is it works, but more compact would look better

Reply:
what we want to avoid is drawing 4 lines when only one semidim is available